recipes/leaf/all/conanfile.py

```diff
@@ -0,0 +1,41 @@
+from conans import tools, ConanFile
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class BoostLeafConan(ConanFile):
+    name = "leaf"
+    description = "Lightweight Error Augmentation Framework"
+    topics = ("conan", "leaf", "boost", "error")
+    license = "BSL-1.0"
+    homepage = "https://github.com/boostorg/leaf"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy(pattern="LICENSE_1_0.txt", src=self._source_subfolder, dst="licenses")
+        self.copy("*", dst="include", src=os.path.join(self._source_subfolder, "include"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "leaf"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "leaf"
+        self.cpp_info.names["cmake_find_package"] = "Boost"
+        self.cpp_info.names["cmake_find_package_multi"] = "Boost"
+        self.cpp_info.components["leaflib"].names["cmake_find_package"] = "leaf"
+        self.cpp_info.components["leaflib"].names["cmake_find_package_multi"] = "leaf"
```

Comment:
targets are  not exported: https://github.com/boostorg/leaf/blob/develop/CMakeLists.txt

Reply:
Leaf is part of Boost and is exported with the pattern `Boost::<lib>` like here https://github.com/boostorg/leaf/blob/develop/CMakeLists.txt#L11.  I just put this to ensure compatibility.

Reply:
I see, it's just that leaf library alone doesn't export its target, there is no official config file.

Reply:
Open a PR with the following (untested) code, get it accepted and we'll happily add the targets.
```cmake
include(GNUInstallDirs)
target_include_directories(boost_leaf PUBLIC
    $<BUILD_INTERFACE:include>
    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDE_DIR}>
)
set_target_properties(boost_leaf PROPERTIES EXPORT_NAME leaf)
option(LEAF_INSTALL "Install leaf" OFF)
if(LEAF_INSTALL)
    install(TARGETS boost_leaf EXPORT leafTargets)
    install(DIRECTORY include/ DESTINATION "${CMAKE_INSTALL_INCLUDE_DIR}")
    install(EXPORT leafTargets DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/leaf/leafConfig.cmake" NAMESPACE "Boost::")
endif()
```

Reply:
if it's a part of boost, is it installed with regular boost package already?

Reply:
Its part of boost since 1.75. But it isn't installed with the regular boost package.